ihe-interop/src/test/java/org/motechproject/ihe/interop/handler/helper/IHEActionHelperTest.java

```diff
@@ -61,4 +66,25 @@ public void shouldThrowExceptionWhenRecipientNotFound() throws ClassNotFoundExce
         when(iheTemplateDataService.findByName("sampleTemplate3")).thenReturn(cdaTemplate);
         iheActionHelper.handleAction(params);
     }
+
+    @Test
+    public void shouldCallSendTemplateMethod() throws IOException, SAXException, ParserConfigurationException, ClassNotFoundException {
+        Map<String, Object> params = new HashMap<>();
+        params.put(Constants.TEMPLATE_NAME_PARAM, "sampleTemplate3");
+        params.put(Constants.RECIPIENT_NAME_PARAM, "sampleRecipient3");
+
+        byte[] bytes = IOUtils.toByteArray(getClass().getResourceAsStream("/empty_template.xml"));
+        Byte[] byteObjects = new Byte[bytes.length];
+        int i = 0;
+        for (byte b : bytes) {
+            byteObjects[i++] = b;
+        }
+
+
+        when(iheTemplateDataService.findByName("sampleTemplate3")).thenReturn(cdaTemplate);
+        when(hl7RecipientsService.getRecipientbyName("sampleRecipient3")).thenReturn(hl7Recipient);
+        when((Byte[]) iheTemplateDataService.getDetachedField(cdaTemplate, "templateData")).thenReturn(byteObjects);
+
+        iheActionHelper.handleAction(params);
+    }
```

Comment:
Maybe use verify() clause to verify if method was called with proper parameters.